presto-tests/src/test/java/com/facebook/presto/tests/TestQueryTaskLimit.java

```diff
@@ -90,16 +90,31 @@ public void testExceedTaskLimit()
     public void testQueuingWhenTaskLimitExceeds()
             throws Exception
     {
-        try (DistributedQueryRunner queryRunner = createQueryRunner(defaultSession, ImmutableMap.of())) {
-            QueryId firstQuery = createQuery(queryRunner, newSession("test", ImmutableSet.of(), null), LONG_LASTING_QUERY);
+        ImmutableMap<String, String> extraProperties = ImmutableMap.<String, String>builder()
+                .put("experimental.spill-enabled", "false")
+                .put("experimental.max-total-running-task-count-to-not-execute-new-query", "2")
+                .build();
+
+        try (DistributedQueryRunner queryRunner = createQueryRunner(defaultSession, extraProperties)) {
+            QueryId firstQuery = createQuery(queryRunner, newSession("test", ImmutableSet.of(), null),
+                    LONG_LASTING_QUERY);
             waitForQueryState(queryRunner, firstQuery, RUNNING);
 
-            queryRunner.getCoordinator().getResourceGroupManager().get().setTaskLimitExceeded(true);
+            // wait for a while, the first query will schedule at lease 2 tasks, so exceed resource group manager's total task limit
+            try {
+                Thread.sleep(500);
```

Comment:
Can you think of a way to check for the query's state?  Because of GC, or other environmental issues, 500 millis may not consistently give the same desired result.

Reply:
It's a great guidance, I will think about how to achieve it.